sdk/storage/Azure.Storage.DataMovement.Files.Shares/src/ShareFilesStorageResourceProvider.cs

```diff
@@ -300,7 +301,10 @@ public StorageResource FromDirectory(Uri directoryUri, ShareFileStorageResourceO
             {
                 CredentialType.None => new ShareDirectoryClient(directoryUri),
                 CredentialType.SharedKey => new ShareDirectoryClient(directoryUri, _getStorageSharedKeyCredential(directoryUri, false)),
-                CredentialType.Token => new ShareDirectoryClient(directoryUri, _getTokenCredential(directoryUri, false)),
+                CredentialType.Token => new ShareDirectoryClient(
+                    directoryUri,
+                    _getTokenCredential(directoryUri, false),
+                    new ShareClientOptions {  ShareTokenIntent = ShareTokenIntent.Backup }),
```

Comment:
Extremely nit, but there appears to be an extra space before

`ShareTokenIntent = ShareTokenIntent.Backup`